### ISSUE TITLE
cron does not create cron_file if it does not exist

### ISSUE THREAD

**Issue by linsomniac** [ansible/ansible-modules-core#4917]
##### ISSUE TYPE
- Bug Report
##### COMPONENT NAME

cron
##### ANSIBLE VERSION

```
ansible 2.1.1.0
  config file = /home/sean/projects/realgoansible/ansible.cfg
  configured module search path = Default w/o overrides

```
##### CONFIGURATION

None
##### OS / ENVIRONMENT

Ubuntu 16.04 64-bit
##### SUMMARY

Using cron module with the cron_file argument will fail if the file named by cron_file argument doesn't already exist.  Leading to needing to use an extra "file: state=touch changed_when: false" sort of stanza.  I wasn't thinking about changed_when, so I added another block to detect if the file didn't exist and only touch then, but reached out on #ansible because I figured I was missing something.  agaffney said "The cron module is probably just missing the logic to create it if it doesn't exist.  That's probably worth creating an issue on github.

Question: If we create the file, do we take owner/group/mode arguments, or create with known perms and use file: to manage it (without needing state=touch, just to specify permissions)?
##### STEPS TO REPRODUCE

Use "cron: [...] cron_file=/tmp/testcron", remove the file /tmp/testcron and run Ansible, it will fail saying that testcron doesn't exist.
##### EXPECTED RESULTS

Creation of the file
##### ACTUAL RESULTS

Error

**Comment by ansibot** [comment]
@dsummersl, @EvanK, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@dsummersl, @EvanK, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by EvanK** [comment]
The issue here is actually [the cron module _trying to read_ the cronfile](https://github.com/ansible/ansible-modules-core/blob/f9f8af9fd660ec70679bcc550a5b2daea18fdaaa/system/cron.py#L241-L251) first, for parsing and diffing.

We just need to add a file existence check to `CronTab.read`, and then the `.write` _should_ handle creating the file

**Comment by EvanK** [comment]
I stand corrected...in my testing, [this try-except](https://github.com/ansible/ansible-modules-core/blob/f9f8af9fd660ec70679bcc550a5b2daea18fdaaa/system/cron.py#L243-L249) is working as expected (aborts reading when nonexistent file raises an `IOError`), and subsequently creates the file on write.

@linsomniac To answer your question: the **cron** module does not take owner/group/mode values, and would write the cronfile as the `ansible_user` (or the `become_user` if `become` is used on the task).

To that end, I wonder if the failure you are seeing is not due to ansible _literally not having permissions to write the file in question_? When you would use the **file** module to touch the file in question, did it have a `become` clause, giving it elevated permissions?

**Comment by ansibot** [comment]
@dsummersl, @EvanK, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by EvanK** [comment]
@ansibot needs_info

**Comment by linsomniac** [comment]
I can't reproduce this now.  At the time I submitted it, I had even worked with some folks on the #ansible IRC channel, and was urged to open this issue.  But I've set up a new test case and I just can't reproduce it.  So I'm going to close it.

I've been strugging with some "file: mode=" issues when using numeric ownership, and I'm wondering if the directory I was creating this in had weird modes because of that.  But that does't explain why adding the "touch" resolved it.  I'm just shooting in the dark here.  Thanks @EvanK for looking into this.
